lib/git_repository/cloning.rb

```diff
@@ -58,14 +58,16 @@ def svn_has_trunk?
 
   # Sets the reference of a local branch
   def reset_branch(old_head_oid, branch, ref)
-    with_head_change old_head_oid do
-      git_exec 'branch', '-f', branch, ref
+    with_head_change(old_head_oid) do
+      git_exec('branch', '-f', branch, ref)
     end
   end
 
   module ClassMethods
     def is_git_repository?(address)
-      !!(exec 'git', 'ls-remote', address)
+      # GIT_ASKPASS is set to the 'true' executable. It simply returns
+      # successfully. This way, no credentials are supplied.
+      !! exec('git', 'ls-remote', '-h', address, 'GIT_ASKPASS' => 'true')
```

Comment:
Do not leave space between `!` and its argument.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.